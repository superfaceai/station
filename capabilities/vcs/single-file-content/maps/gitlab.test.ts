import { SuperfaceTest } from '@superfaceai/testing-lib';

describe(`vcs/single-file-content/gitlab`, () => {
  let superface: SuperfaceTest;

  beforeEach(() => {
    superface = new SuperfaceTest();
  });

  describe('SingleFileContent', () => {
    it('should perform successfully', async () => {
      await expect(
        superface.run({
          profile: 'vcs/single-file-content',
          provider: 'gitlab',
          useCase: 'SingleFileContent',
          input: {
            owner: 'Jakub-Vacek',
            repo: 'empty-test',
            path: 'README.md',
            ref: 'master',
          },
        })
      ).resolves.toMatchSnapshot();
    });

    it('should map error', async () => {
      await expect(
        superface.run({
          profile: 'vcs/single-file-content',
          provider: 'gitlab',
          useCase: 'SingleFileContent',
          input: {
            owner: 'Jakub-Vacek',
            repo: 'empty-test',
            path: 'README.md',
            ref: 'made-up',
          },
        })
      ).resolves.toMatchSnapshot();
    });
  });
});
