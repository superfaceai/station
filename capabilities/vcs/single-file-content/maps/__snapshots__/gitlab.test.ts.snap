// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vcs/single-file-content/gitlab SingleFileContent should map error 1`] = `
Object {
  "error": "HTTPError: Expected HTTP error
AST Path: definitions[0].statements[3].responseHandlers[0].statements[1]
Original Map Location: Line 19, column 7",
}
`;

exports[`vcs/single-file-content/gitlab SingleFileContent should perform successfully 1`] = `
Object {
  "value": Object {
    "content": "MQojIGVtcHR5LXRlc3QKdGhpcyBpcyBlbXB0eSByZXBvCg==",
    "encoding": "base64",
    "size": 34,
  },
}
`;
